File: previous_versions/v0dot2/project_environment_all/py_unittest/test_collect_coins.py
import unittest
from environment_all import GenericWorld

class TestCollectCoins(unittest.TestCase):
    def test_collect_coins(self):
        # Arrange
        world = GenericWorld()  # Create an instance of the GenericWorld class
        coin = Coin()  # Create a Coin instance
        agent = Agent()  # Create an Agent instance
        
        # Set the necessary properties for the coin and agent
        coin.collectable = True
        coin.x = agent.x
        coin.y = agent.y
        agent.score = 0
        agent.events = []
        agent.trophies = []
        
        # Add the coin and agent to the world
        world.coins = [coin]
        world.active_agents = [agent]
        
        expected_log = f"Agent <{agent.name}> picked up coin at {(agent.x, agent.y)} and receives 1 point"
        
        # Act
        world.collect_coins()  # Call the collect_coins method
        
        # Assert
        self.assertFalse(coin.collectable)  # Check that the 'collectable' property of the coin is set to False
        self.assertEqual(agent.score, 1)  # Check that the agent's score has been updated to 1
        self.assertIn(e.COIN_COLLECTED, agent.events)  # Check that the COIN_COLLECTED event has been added to the agent's events
        self.assertIn(Trophy.coin_trophy, agent.trophies)  # Check that the coin_trophy has been added to the agent's trophies
        self.assertIn(expected_log, world.logger.info.call_args[0][0])  # Check that the expected log message has been logged

if __name__ == '__main__':
    unittest.main()

File: previous_versions/v0dot2/project_environment_all/js_unittest/test_collect_coins.js
JavaScript unit test:

const GenericWorld = require('./EnvironmentAll').GenericWorld;

describe('CollectCoins', () => {
    test('collect_coins', () => {
        // Arrange
        const world = new GenericWorld();  // Create an instance of the GenericWorld class
        const coin = new Coin();  // Create a Coin instance
        const agent = new Agent();  // Create an Agent instance
        
        // Set the necessary properties for the coin and agent
        coin.collectable = true;
        coin.x = agent.x;
        coin.y = agent.y;
        agent.score = 0;
        agent.events = [];
        agent.trophies = [];
        
        // Add the coin and agent to the world
        world.coins = [coin];
        world.active_agents = [agent];
        
        const expected_log = `Agent <${agent.name}> picked up coin at (${agent.x}, ${agent.y}) and receives 1 point`;
        
        // Act
        world.collect_coins();  // Call the collect_coins method
        
        // Assert
        expect(coin.collectable).toBe(false);  // Check that the 'collectable' property of the coin is set to false
        expect(agent.score).toBe(1);  // Check that the agent's score has been updated to 1
        expect(agent.events).toContain(e.COIN_COLLECTED);  // Check that the COIN_COLLECTED event has been added to the agent's events
        expect(agent.trophies).toContain(Trophy.coin_trophy);  // Check that the coin_trophy has been added to the agent's trophies
        expect(world.logger.info.mock.calls[0][0]).toContain(expected_log);  // Check that the expected log message has been logged
    });
});